lightning/src/ln/peers/handler.rs

```diff
@@ -30,9 +29,58 @@ use std::ops::Deref;
 use bitcoin::hashes::sha256::Hash as Sha256;
 use bitcoin::hashes::sha256::HashEngine as Sha256Engine;
 use bitcoin::hashes::{HashEngine, Hash};
-use ln::peers::handshake::PeerHandshake;
-use ln::peers::conduit::Conduit;
-use ln::peers::handshake::acts::Act;
+use ln::peers::outbound_queue::OutboundQueue;
+use ln::peers::transport::{PayloadQueuer, Transport};
+use bitcoin::hashes::core::iter::Filter;
+use std::collections::hash_map::IterMut;
+
+const MSG_BUFF_SIZE: usize = 10;
+
+/// Interface PeerManager uses to interact with the Transport object
+pub(super) trait ITransport: MessageQueuer {
+	/// Instantiate the new outbound Transport
+	fn new_outbound(initiator_static_private_key: &SecretKey, responder_static_public_key: &PublicKey, initiator_ephemeral_private_key: &SecretKey) -> Self;
+
+	/// Set up the Transport receiving any bytes that need to be sent to the peer
+	fn set_up_outbound(&mut self) -> Vec<u8>;
+
+	/// Instantiate a new inbound Transport
+	fn new_inbound(responder_static_private_key: &SecretKey, responder_ephemeral_private_key: &SecretKey) -> Self;
+
+	/// Process input data similar to reading it off a descriptor directly.
+	fn process_input(&mut self, input: &[u8], output_buffer: &mut impl PayloadQueuer) -> Result<bool, String>;
+
+	/// Returns true if the connection is established and encrypted messages can be sent.
+	fn is_connected(&self) -> bool;
+
+	/// Returns the node_id of the remote node. Panics if not connected.
+	fn get_their_node_id(&self) -> PublicKey;
+
+	/// Returns all Messages that have been received and can be parsed by the Transport
+	fn drain_messages<L: Deref>(&mut self, logger: L) -> Result<Vec<Message>, PeerHandleError> where L::Target: Logger;
+}
+
+/// Interface PeerManager uses to queue message to send. Used primarily to restrict the interface in
+/// specific contexts. e.g. Only queueing during read_event(). No flushing allowed.
+pub(super) trait MessageQueuer {
```

Comment:
Should the trait by Send + Sync ? You might be willingly to implement a threading pool behind this interface, serialization and encryption are likely performance points.

Reply:
I had the same thought and came to the conclusion that until we know how it is used we can just defer this decision. I fully expect performance work to end up in this layer, but until we know what we need and how it fits into the existing object model, I think it is better to leave it as-is.